### DISCUSSION TITLE
Reload Hangs - Inconsistent

### DISCUSSION THREAD

**Discussion by JohnFromGWN** [scripthookvdotnet/scripthookvdotnet#1488] [Q&A]
First of all, to preface, I've been using SHVDN for years.  Lately hitting the reload key (F11 in my case) behaves inconsistently. This happens when I have visual studio open, i rebuild directly to the scripts folder (or subfolder in my case). The reload can take a few seconds or it can hang for close to a minute, often longer. It will not crash the game, just hangs.

Once more this is inconsistent and super annoying when modifying or testing scripts. At times I need to ctrl esc multiple times to get the game to respond. I don't use any 3rd party scripts, definitely no missions or complex scripts, and the majority of my scripts are LemonUI based and a few custom camera scripts. 

So my question is. Is there a way to troubleshoot this without enabling and disabling scripts? That would be my first choice but as I wrote above the behaviour is inconsistent. Sure I could go back to the last scripts i added as they are most likely to blame, but I'm asking if there is more efficient way to identify what's causing this recent phenomenon.

I'm on a super stable setup with 2372 reverted version and SHVDN 3.6 release (not the nightly) and latest LemonUI. Nothing has changed in my game or environment except adding new scripts.

Thanks

**Comment by kagikn** [comment]
I don't think there's a way to troubleshoot the hangs, since it may be responsible to SHVDN's implementation.
I'm not sure how reloading can hang for like a minute, but you could try nightly versions for more consistent runtime behaviors. You might want to know that some nightly versions *can* work stably than the latest stable version when the user have installed, even without considering #1451.

I don't really remember if I saw users that claimed SHVDN v3.6.0 inconsistently behaved when reloading scripts, before the game version 1.0.3258.0 was released. However, I'm sure at least all the stable version between v3.0.0 and v3.6.0 and all the nightly versions prior to nightly.98 do have a large amount of factors that may cause race conditions in the runtime that cause it to inconsistently behave. crosire didn't appear to know that keyboard event functions run in a different thread from tick functions.
